x/gamm/types/marshal.go

```diff
@@ -0,0 +1,127 @@
+package types
+
+import (
+	"encoding/json"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	types "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"gopkg.in/yaml.v2"
+)
+
+type poolAssetPretty struct {
+	Token  types.Coin `json:"token" yaml:"token"`
+	Weight sdk.Dec    `json:"weight" yaml:"weight"`
+}
+
+func (asset PoolAsset) prettify() poolAssetPretty {
+	return poolAssetPretty{
+		Weight: sdk.NewDecFromInt(asset.Weight).QuoInt64(GuaranteedWeightPrecision),
+		Token:  asset.Token,
+	}
+}
+
+// D: at name
+// func (asset poolAssetPretty) uglify() PoolAsset {
```

Comment:
Remove code?

Reply:
I think its useful to show the reverse map in case we need an UnmarshalYAML someday, wdyt

Reply:
Ok